src/redux/addCash.js

```diff
@@ -69,6 +76,65 @@ export const addCashNewPurchaseTransaction = txDetails => (
   savePurchaseTransactions(updatedPurchases, accountAddress, network);
 };
 
+export const addCashGetTransferHash = (
+  referenceInfo,
+  transferId,
+  sourceAmount
+) => async (dispatch, getState) => {
+  const { accountAddress, network } = getState().settings;
+  const { assets } = getState().data;
+  const getTransferHash = async (referenceInfo, transferId, sourceAmount) => {
+    try {
+      const {
+        destAmount,
+        destCurrency,
+        transferHash,
+      } = await trackWyreTransfer(referenceInfo, transferId, network);
+
+      const destAssetAddress = toLower(
+        AddCashCurrencies[network][destCurrency]
+      );
+
+      if (transferHash) {
+        logger.log('Wyre transfer hash', transferHash);
+        let asset = ethereumUtils.getAsset(assets, destAssetAddress);
+        if (!asset) {
+          asset = AddCashCurrencyInfo[network][destAssetAddress];
+        }
+        const txDetails = {
+          amount: destAmount,
+          asset,
+          from: null,
+          hash: transferHash,
+          nonce: null,
+          sourceAmount,
+          status: TransactionStatusTypes.purchasing,
+          timestamp: Date.now(),
+          to: accountAddress,
+          transferId,
+          type: TransactionTypes.purchase,
+        };
+        const newTxDetails = await dispatch(
+          dataAddNewTransaction(txDetails),
+          false
+        );
+        dispatch(addCashNewPurchaseTransaction(newTxDetails));
+      } else {
+        setTimeout(
+          () => getTransferHash(referenceInfo, transferId, sourceAmount),
+          1000
+        );
+      }
+    } catch (error) {
+      setTimeout(
```

Comment:
In this scenario, do you wanna retry forever?

Reply:
Good point! I'm going to have a separate PR to manage places where we have "infinite" retries to cap them at a certain limit